src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -1083,68 +1085,40 @@ private RegexNode ScanReplacement()
          */
         private void ScanBlank()
         {
-            if (UseOptionX())
+            while (true)
             {
-                while (true)
+                if (UseOptionX())
```

Comment:
This change means UseOptionX is now being checked inside rather than outside of the loop, and potentially multiple times.

Reply:
I figured it's negligible - I can extract to a bool? Or you would rather the loop be blown up as it was?

Reply:
> I figured it's negligible

Did you measure?